### PR TITLE
Refactor codebase

### LINKED ISSUE #8: Refactor
I think it would be cleaner now if the code was split up into a `lib` folder for each OS:

- `lib/macos.js`
- `lib/linux.js`
- `lib/windows.js`

Where each expose an interface of `exports.async` and `exports.sync`. `index.js` would then *only* be responsible for delegating to the correct OS functions.

### PR DESCRIPTION
**Pull request by itaisteinherz**
Splitted up the code in `index.js` to 3 separate files for each OS. This closes #8.

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -1,90 +1,30 @@
...
-	};
-};
-
-const getActiveWindowId = activeWindowIdStdout => parseInt(activeWindowIdStdout.split('\t')[1], 16);
 
 module.exports = () => {
 	if (process.platform === 'darwin') {
-		return pify(childProcess.execFile)(bin).then(parseMac);
+		const macos = require('./lib/macos');
+		return macos();
```
Comment (sindresorhus) [inline index.js]: Let's simplify all these to `return require('./lib/macos')()`.

Comment (itaisteinherz) [inline index.js]: Should I use that style for the `.sync` function as well?

Comment (sindresorhus) [inline index.js]: Yes, just `return require('./lib/macos').sync();`

**Comment by sindresorhus** [comment]
You also need to update https://github.com/sindresorhus/active-win/blob/69bb7bd52d6b8ebc8c290de5c469d100a81981cb/package.json#L19-L23

**Comment by itaisteinherz** [comment]
Done 👍

**Comment by sindresorhus** [comment]
@itaisteinherz Could you fix the merge conflict? We had to do an important change in the Windows code.

**Comment by itaisteinherz** [comment]
Sure 👍

File: package.json [hunk 2]
```diff
@@ -18,7 +18,9 @@
   },
   "files": [
     "index.js",
-    "windows.js",
+    "lib/linux.js",
+    "lib/macos.js",
+    "lib/windows.js",
```
Comment (sindresorhus) [inline package.json]: You can just add the whole folder here:

```
"lib",
```

**Comment by sindresorhus** [comment]
@itaisteinherz Can you give this a quick look?

File: package.json (hunk 2 above)
Comment (itaisteinherz) [inline package.json]: Oh right, my bad 😅

**Comment by itaisteinherz** [comment]
Sure

**Comment by sindresorhus** [comment]
Thank you @itaisteinherz :)

**Pull request merged**